File: API/src/main/java/com/chamados/api/Entities/Department.java
package com.chamados.api.Entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.util.Objects;

@Getter
@Entity
@Table(name="departments")
public class Department {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Setter
    private String name;

	@Setter
	private boolean receivesRequests;
	
	@ManyToOne
	@JoinColumn(name = "unit_id")
	@Setter
    private Unit unit;
	
	public Department() {}
	
	public Department(String name, boolean receivesRequests, Unit unit) {
		this.name = name;
		this.receivesRequests = receivesRequests;
		this.unit = unit;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		Department that = (Department) o;
		return Objects.equals(id, that.id);
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}
}

File: frontend/src/types/Department.ts
export class Department {
    id: string | null;
    name: string;
    unit: { id: string; name: string; address: string };
    receivesRequests: boolean;

    constructor(
        id = "",
        name = "",
        unit = { id: "", name: "", address: "" },
        receivesRequests = false
    ) {
        this.id = id;
        this.name = name;
        this.unit = unit;
        this.receivesRequests = receivesRequests;
    }
}
